src/VAC/VectorAnimationComplex/InfillPattern.cpp

```diff
@@ -0,0 +1,473 @@
+                infillPoints << potentialRightPoint;
+                infillPoints << potentialLeftPoint;
+            } else {
+                infillPoints << potentialLeftPoint;
+                infillPoints << potentialRightPoint;
+            }
+        }
+        directionRight = !directionRight;
+    }
+
+    data_ << QVector<QVector<QPointF>>{infillPoints};
+}
+
+inline QPointF rotateAndNormalizePoint(double angle, const QPointF& vector)
+{
+    constexpr auto r = 1;
+    auto phi = atan2(vector.y(), vector.x());
+    phi += angle;
+    return QPointF{r * cos(phi), r * sin(phi)};
+}
+
+QPointF lineLineIntersection(QPointF A, QPointF B, QPointF C, QPointF D)
+{
+    // Line AB represented as a1x + b1y = c1
+    double a1 = B.y() - A.y();
+    double b1 = A.x() - B.x();
+    double c1 = a1*(A.x()) + b1*(A.y());
+
+    // Line CD represented as a2x + b2y = c2
+    double a2 = D.y() - C.y();
+    double b2 = C.x() - D.x();
+    double c2 = a2*(C.x())+ b2*(C.y());
+
+    double determinant = a1*b2 - a2*b1;
+
+    if (determinant == 0)
+    {
+        // The lines are parallel. This is simplified
+        // by returning a pair of FLT_MAX
+        return QPointF(FLT_MAX, FLT_MAX);
+    }
+    else
+    {
+        double x = (b2*c1 - b1*c2)/determinant;
+        double y = (a1*c2 - a2*c1)/determinant;
+        return QPointF(x, y);
+    }
+}
+
+void InfillPattern::gridInfill()
+{
+    rectiLinearHorizontalInfill();
+    const auto outerBoundingBox = inputPolygon_.boundingRect();
+    const auto startBoundary = outerBoundingBox.left();
+    const auto endBoundary = outerBoundingBox.right();
+    const auto top = outerBoundingBox.top();
+    const auto bottom = outerBoundingBox.bottom();
+
+    for (auto x = startBoundary; x <= endBoundary; x += spacing_) {
+        auto potentialTopPoint = QPointF{x, top};
+        while(!inset_.containsPoint(potentialTopPoint, Qt::OddEvenFill) && potentialTopPoint.y() <= bottom) {
+            potentialTopPoint += QPointF{0, 1};
+        }
+        auto potentialBottomPoint = QPointF{x, bottom};
+        while(!inset_.containsPoint(potentialBottomPoint, Qt::OddEvenFill) && potentialBottomPoint.y() >= top) {
+            potentialBottomPoint -= QPointF{0, 1};
+        }
+
+        if (potentialBottomPoint.y() - potentialTopPoint.y() > spacing_ &&
+                inset_.containsPoint(potentialTopPoint, Qt::OddEvenFill) &&
+                inset_.containsPoint(potentialBottomPoint, Qt::OddEvenFill)) {
+            QVector<QPointF> verticalLine;
+            verticalLine << potentialBottomPoint << potentialTopPoint;
+            data_ << verticalLine;
+        }
+    }
+}
+
+void InfillPattern::concentricInfill()
+{
+    const auto insetSpacing = spacing_ / 2;
+    auto currentBorder = inset_;
+    while(currentBorder.boundingRect().right() - currentBorder.boundingRect().left() > insetSpacing &&
+          currentBorder.boundingRect().bottom() - currentBorder.boundingRect().top() > insetSpacing) {
```

Comment:
add parenthesis around subtraction to make order-of-operations a bit clearer

Reply:
Well then the subtraction would be between a double and bool which doesn't make sense... I think it's clearer like this but I've added extra parentheses

Reply:
? semantics should remain the same

just change `if( a - b > s && c - d > s)` to `if ((a-b)>s && (c-d) > s)`

Reply:
Yeah sorry I explained it badly. I mean if it didn't work the correct way there would be subtraction between a double and bool. What I meant is that there is not ambiguity in this case since it can only work one way (in comparison to * and / where the order could possibly be either way [although of course there is a specified ordering in C++]).